src/aqua/decider/mailbox.aqua

```diff
@@ -0,0 +1,54 @@
+module Mailbox declares poll_mailbox, push_remove_worker
+
+export poll_mailbox, push_remove_worker
+
+import Spell from "@fluencelabs/spell/spell_service.aqua"
+import PeerId from "@fluencelabs/aqua-lib/builtin.aqua"
+import Worker from "@fluencelabs/aqua-lib/workers.aqua"
+
+import Console from "../debug.aqua"
+import SpellId, WorkerId from "../types.aqua"
+
+import spell_log from "../fluence/spell.aqua"
+
+data RemoveWorker:
+    host_id: PeerId
+    worker_id: WorkerId
+
+data DeciderMessage:
+    remove: ?RemoveWorker
+
+service JsonMsg("json"):
+  parse(str: string) -> DeciderMessage
+  stringify(msg: DeciderMessage) -> string
+
+func remove_worker(worker: RemoveWorker):
+    spell_log("spell", ["will remove worker", worker.worker_id, "from host", worker.host_id])
+    on worker.host_id:
+        Worker.remove(worker.worker_id)
+
+func process_message(msg: DeciderMessage):
+    if msg.remove != nil:
+        remove_worker(msg.remove!)
+
+func poll_mailbox(spell_id: SpellId):
+    Spell spell_id
+
+    mailbox <- Spell.get_mailbox()
+    spell_log(spell_id, ["mailbox", mailbox])
+    if mailbox.success:
+        for msg <- mailbox.messages:
+            try:
+                message <- JsonMsg.parse(msg)
+                process_message(message)
+                spell_log(spell_id, "will pop")
+                popped <- Spell.pop_mailbox()
+                -- if popped.str != msg:
+                --     spell_log(spell_id, ["произошла чудовищная ошибка. expected", msg, "popped", popped])
+
+func push_remove_worker(spell_id: SpellId, host_id: PeerId, worker_id: WorkerId):
```

Comment:
```suggestion
func request_remove_worker(spell_id: SpellId, host_id: PeerId, worker_id: WorkerId):
```
I prefer this naming because you don't interact with mailbox directly and "push" sounds too vague

Reply:
push to mailbox, seems very direct to me.

but you're right that we miss a bigger abstraction for this. we'll come to that :)